README.md

```diff
@@ -304,6 +306,54 @@ This will not work:
 
 	sslh --listen 192.168.0.1:443 --ssh 127.0.0.1:22 --ssl 127.0.0.1:4443
 
+
+Systemd Socket Activation
+-------------------------
+If compiled with `USESYSTEMD` then it is possible to activate 
+the service on demand and avoid running any code as root.
+
+In this mode any listen configuration options are ignored and 
+the sockets are passed by systemd to the service.
+
+Example socket unit:
+
+	[Unit]
+	Before=sslh.service
+	
+	[Socket]
+	ListenStream=1.2.3.4:443
+	ListenStream=5.6.7.8:444
+	ListenStream=9.10.11.12:445
+	FreeBind=true
+
+	[Install]
+	WantedBy=sockets.target
+
+Example service unit:
+
+	[Unit]
+	PartOf=sslh.socket
+	
+	[Service]
+	ExecStart=/usr/sbin/sslh -v -f --ssh 127.0.0.1:22 --ssl 127.0.0.1:443
+	KillMode=process
+	CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_NET_ADMIN CAP_SETGID CAP_SETUID
```

Comment:
Indeed no capabilities are needed as the socket is all set up in this
scenario...

However the sockets are optional. In the event systemd sockets have decided
not to be used by the administrator caps are then needed.

The idea is to provide flexibility... I'm unsure yet how I'll go with the
Fedora and EPEL7 default behaviour...

I'm also in the process of writing a systemd generator to parse the
sslh.cfg and generate socket files from that.
On 19 Aug 2015 23:06, "Thomas Weißschuh" <notifications@github.com> wrote:

> In README.md
> <https://github.com/yrutschle/sslh/pull/56#discussion_r37474816>:
>
> > +	ListenStream=5.6.7.8:444
> > +	ListenStream=9.10.11.12:445
> > +	FreeBind=true
> > +
> > +	[Install]
> > +	WantedBy=sockets.target
> > +
> > +Example service unit:
> > +
> > +	[Unit]
> > +	PartOf=sslh.socket
> > +	
> > +	[Service]
> > +	ExecStart=/usr/sbin/sslh -v -f --ssh 127.0.0.1:22 --ssl 127.0.0.1:443
> > +	KillMode=process
> > +	CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_NET_ADMIN CAP_SETGID CAP_SETUID
>
> The capabilieties could be dropped in this case, no?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yrutschle/sslh/pull/56/files#r37474816>.
>